src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlStringMethodTranslator.cs

```diff
@@ -334,6 +338,7 @@ public NpgsqlStringMethodTranslator(NpgsqlTypeMappingSource typeMappingSource, I
                 || method == String_Join2
                 || method == String_Join3
                 || method == String_Join4
+                || (method == String_Join5 && arguments[1].TypeMapping is NpgsqlArrayTypeMapping)
```

Comment:
The check for NpgsqlArrayTypeMapping should also be applied for the other String_Join overloads, no? The same problem (e.g. value-converted array) could occur there as well.

Reply:
I first thought they'd just compile for stuff that works, but the value-converted array is a good counter example. I'll move the check one level up and add the test for the array case.